python/lsst/ts/externalscripts/auxtel/latiss_acquire_and_take_sequence.py

```diff
@@ -245,116 +285,371 @@ def set_metadata(self, metadata):
             gratings.add(grating)
         metadata.filter = f"{filters},{gratings}"
 
-    async def latiss_acquire(self, silent=True, doPointingModel=False):
+    async def get_next_image_data_id(self, timeout=STD_TIMEOUT, flush=True):
+        """Return dataID of image that appears from the ATArchiver CSC.
+        This is meant to be called at the same time as a take image command.
+        If this is called after take_image is completed, it may not receive
+        the imageInOODS event.
+
+        Inputs:
+        timeout: `float`
+            Amount of time to wait for image to arrive.
+        """
+
+        self.log.info(
+            f"Waiting for image to arrive in OODS for a maximum of {timeout} seconds."
+        )
+        in_oods = await self.latiss.rem.atarchiver.evt_imageInOODS.next(
+            timeout=timeout, flush=flush
+        )
+
+        day_obs, seq_num = parse_obs_id(in_oods.obsid)[-2:]
+        self.log.info(f"seqNum {seq_num} arrived in OODS")
+
+        data_id = dict(dayObs=day_obs, seqNum=seq_num)
+
+        return data_id
+
+    async def latiss_acquire(self, doPointingModel=False):
 
         if doPointingModel:
-            targetPosition = boreSight
+            target_position = latiss_constants.boresight
         else:
-            targetPosition = sweetSpots[self.acq_grating]
+            target_position = latiss_constants.sweet_spots[self.acq_grating]
+
+        # get filter/disperser currently in the beam
+        (
+            current_filter,
+            current_grating,
+            current_stage_pos,
+        ) = await self.latiss.get_setup()
+
+        # Is the atspectrograph Correction in the ATAOS running?
+        corr = await self.atcs.rem.ataos.evt_correctionEnabled.aget(
+            timeout=self.cmd_timeout
+        )
 
-        # Automatically accept calculated offset to sweetspot?
-        # if False this queries the user, which will fail badly on
-        # the script queue so make it always True
-        _alwaysAcceptMove = True
+        # Setup instrument and telescope
+        # Following code requires persistent offsets to be functional, but
+        # that's not yet the case therefore this will result in a race
+        # condition where the pointing offsets might be wiped out.
+        # await asyncio.gather(
+        #     self.atcs.slew_object(name=self.object_name, rot=0,
+        #     slew_timeout=240),
+        #     self.latiss.setup_instrument(grating=self.acq_grating,
+        #     filter=self.acq_filter),
+        # )
+
+        # Slew and setup in series for now
+        # set rot_type to align to paralactic angle
+        await self.atcs.slew_object(name=self.object_name, rot_type=2, slew_timeout=240)
```

Comment:
Consider importing the enumeration and using it here:

At top do:
```
from lsst.ts.observatory.control.utils import RotType
```

Then here do:

```
await self.atcs.slew_object(name=self.object_name, rot_type=RotType.Parallactic., slew_timeout=240)

```

The instrument setup is done later (in line 70), but the code you comment above is doing it in parallel with `asyncio.gather`... May be a good idea to either do them in sequence here:

```
await self.atcs.slew_object(name=self.object_name, rot_type=2, slew_timeout=240)
await self.latiss.setup_instrument(grating=self.acq_grating, filter=self.acq_filter),
```

or leave a comment here so you remember to remove the code in line 370 once persistent offset it working.

Reply:
i'm confused... the gather statement there is commented out because of the persistent offset issue...
I added a comment, does that address the issue?